src/cargo/util/command_prelude.rs

```diff
@@ -986,6 +1003,64 @@ pub fn root_manifest(manifest_path: Option<&Path>, gctx: &GlobalContext) -> Carg
     }
 }
 
+pub fn lockfile_path(
+    lockfile_path: Option<&Path>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<PathBuf>> {
+    let path;
+
+    if let Some(lockfile_path) = lockfile_path {
```

Comment:
Use `let-else` so we can have less indentation levels.

```suggestion
    let Some(lockfile_path) = lockfile_path else {
        return Ok(None);
    };
```